src/accelerate/test_utils/scripts/test_script.py

```diff
@@ -540,6 +541,19 @@ def training_check(use_seedable_sampler=False):
         assert torch.allclose(old_model.b, model.b), "Did not obtain the same model on XPU or distributed training."
 
 
+def test_split_between_processes_dataset():
+    assert is_datasets_available(), "test requires Hugging Face datasets"
+    from datasets import Dataset
+
+    state = AcceleratorState()
+    data = Dataset.from_list([dict(k=v) for v in range(2 * state.num_processes)])
+    with state.split_between_processes(data) as results:
+        assert (
+            len(results) == 2
```

Comment:
> We only test for if it can be evenly split. In Accelerate we need to make sure we cover bases when evenly split is not an option. Can you write a test for an odd number of dataset items?

done

>If that test fails, you will need to do the padding logic similar to what we do.

done

> Also this needs to be added down under if __name__ == "__main__" where the other test_split_x is to actually be called.

ok

> If we really wanted to get nitty/gritty, down there is where we can also include is_datasets_available() before calling the test function instead of having the assert here.

also done but it's ugly. because of scope, an import in `main()` is not accessible in `test_split_between_processes_dataset()` that's why we now pass the imported `datasets.Dataset` as a parameter to  `test_split_between_processes_dataset()` 

**one more thing**, with small datasets very counter-intuitive things happen because of the `math.ceil` in `num_samples_per_process = math.ceil(length / self.num_processes)`.

For example
```
Num. of GPUs: 5,
Num. of Samples: 6 ([1, 2, 3, 4, 5, 6])
Apply padding: False
GPU0: (['1', '2'])	GPU1: (['3', '4'])	GPU2: (['5', '6'])	GPU3: (['6'])	GPU4: (['6'])
```

but the same happens with `split_between_processes` for a `list` so I guess that's an accepted behaviour.